### DISCUSSION TITLE
Blade directive @can('users.index') doesn't works

### DISCUSSION THREAD

**Discussion by miguel-cortez-weedmaps** [spatie/laravel-permission#1740] [Q&A]
Hello I have a question with blade directive @can('permission')

I have a user that has a role admin, this roles has permissions like 'users.index', so when I trying to use 
@can('users.index')
< li>< a class="active" href="/admin/users">Users< /a>< /li>
@endcan
The link is not showing, but when I use @ role('admin'), so by role it works. 

I'm using Laravel 8 and I cleared cache with 
php artisan config:clear
php artisan permission:cache-reset. 

I hope you can help me please.

**Comment by drbyte** [comment]
The `@can()` directive is driven by Laravel's internals, not by this package. That is, when `@can` parses things, that's Laravel parsing it, not this package. (This package just registers the names of the current user's permissions so that `@can` is aware of them.)
IIRC laravel's internals treat the `.` with special handling, so that might be confusing things. 
You could try using `auth()->user()->hasPermissionTo()`.

**Reply by miguel-cortez-weedmaps** [reply]
I know @can() is of Laravel but it doesn't recognizes permission name

I'm trying auth()->user()->hasPermissionTo()
I got this: 

Spatie\Permission\Exceptions\PermissionDoesNotExist
There is no permission named `users.index` for guard `web`

In my database i have 
2	users.index	        	2021-05-13 17:10:02	2021-05-13 17:10:13

**Reply by drbyte** [reply]
If you're using a different guard than `web`, you can pass the guard name to `hasPermissionTo`
